### ISSUE TITLE
Cannot run single executeable app

### ISSUE THREAD

**Issue by AndnixSH** [dotnet/runtime#37220]
### Description

I'm compiling a project called Il2CppInspector without touching anything https://github.com/djkaty/Il2CppInspector
It does have an instruction how to build and I followed the steps excatly
but I'm getting an error when trying to execute it:

```
Failure processing application bundle.
Bundle header version compatibility check failed
A fatal error was encountered. Could not extract contents of the bundle
```

I tried to upgrade to .NET Core 3.1, same issue

Executing normal app which have all the assemblies outside works perfectly as it should but I want to release single executeable app

### Configuration

My OS is Win10 x64. The project is based on .NET Core 3.0 and compiles for x64 by default

**Comment by swaroop-sridhar** [comment]
@AndnixSH thanks for reporting the issue. Which version of the SDK are you using the build the app?

**Comment by AndnixSH** [comment]
The SDK usally installed automatically when I install Visual Studio 2019 I guess...?

Oh well, I had the preview version 5.0.100-preview.3.20216.6 installed which might be the reason why. I don't even know i had it installed or why i had installed it, I should have checked

I uninstalled .NET Core 5 preview, keep the 3.1.300. re-compiled the project and it works now

Because I open the issue, I decided to update to 5.0.100-preview.4.20258.7 to try again. The single executeable app works, so i think this issue has been solved
